src/screens/Profiles/UserLibrary/components/UserLibraryScreen/component.js

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react';
+import { Container, Icon } from 'native-base';
+import { FlatList, Image, ScrollView, StyleSheet, TouchableOpacity, View } from 'react-native';
+import PropTypes from 'prop-types';
+import ScrollableTabView from 'react-native-scrollable-tab-view';
+import { ProfileHeader } from 'kitsu/components/ProfileHeader';
+import { LibraryHeader } from 'kitsu/screens/Profiles/UserLibrary';
+import { ScrollableTabBar } from 'kitsu/components/ScrollableTabBar';
+import { SearchBar } from 'kitsu/components/SearchBar';
+import { ProgressBar } from 'kitsu/components/ProgressBar';
+import { styles } from './styles';
+
+const MINIMUM_SEARCH_TERM_LENGTH = 3;
+
+export class UserLibraryScreenComponent extends React.Component {
+  static navigationOptions = (props) => {
+    const { profile } = props.navigation.state.params;
+
+    return {
+      headerStyle: {
+        shadowColor: 'transparent',
+        elevation: 0,
+      },
+      header: () => (
+        <ProfileHeader
+          profile={profile}
+          title={profile.name}
+          onClickBack={props.navigation.goBack}
+        />
+      ),
+      tabBarIcon: ({ tintColor }) => (
+        <Icon ios="ios-body" android="md-body" style={{ fontSize: 24, color: tintColor }} />
+      ),
+    };
+  };
+
+  constructor() {
+    super();
+
+    this.state = {
+      searchTerm: '',
+    };
+  }
+
+  componentDidMount() {
+    const { profile } = this.props.navigation.state.params;
+    this.props.fetchUserLibrary(profile.id);
+  }
+
+  onSearchTermChanged = (searchTerm) => {
+    this.setState({ searchTerm });
+    const { profile } = this.props.navigation.state.params;
+    const { userLibrary } = this.props;
+    const isSearching = userLibrary.searchTerm.length !== 0;
+
+    if (searchTerm.length >= MINIMUM_SEARCH_TERM_LENGTH && !isSearching) {
```

Comment:
Does the `!isSearching` basically try to debounce the search action? If so, you can use lodash's debounce, which might be nicer for the user?